src/coreclr/jit/ee_il_dll.cpp

```diff
@@ -1122,9 +1122,39 @@ void Compiler::eeDispLineInfos()
  * (e.g., host AMD64, target ARM64), then VM will get confused anyway.
  */
 
+void Compiler::eeAllocMem(AllocMemArgs* args, UNATIVE_OFFSET hotSizeRequest, UNATIVE_OFFSET coldSizeRequest)
+{
+#ifdef DEBUG
+    // Fake splitting implementation: hot section = hot code + 4K buffer + cold code
+    const UNATIVE_OFFSET buffer = 4096;
+    if (JitConfig.JitFakeProcedureSplitting() && (coldSizeRequest > 0))
+    {
+        args->hotCodeSize  = hotSizeRequest + buffer + coldSizeRequest;
+        args->coldCodeSize = 0;
+    }
+#endif
+
+    info.compCompHnd->allocMem(args);
+
+#ifdef DEBUG
+    // Fix up hot/cold code pointers
+    if (JitConfig.JitFakeProcedureSplitting() && (coldSizeRequest > 0))
+    {
+        args->coldCodeBlock   = ((BYTE*)args->hotCodeBlock) + hotSizeRequest + buffer;
+        args->coldCodeBlockRW = ((BYTE*)args->hotCodeBlockRW) + hotSizeRequest + buffer;
+    }
+#endif
+}
+
 void Compiler::eeReserveUnwindInfo(bool isFunclet, bool isColdCode, ULONG unwindSize)
 {
 #ifdef DEBUG
+    // Fake splitting currently does not handle unwind info for cold code
+    if (isColdCode && JitConfig.JitFakeProcedureSplitting())
+    {
+        return;
+    }
+
```

Comment:
It might be better to move this immediately before the call to `reserveUnwindInfo`, below, and add:
```
JITDUMP("reserveUnwindInfo for cold code with JitFakeProcedureSplitting enabled: ignoring cold unwind info\n");
```
which would add a line to the JitDump in this case, but also let the normal `reserveUnwindInfo` printf be executed first.